src/optimal_image/search.rs

```diff
@@ -0,0 +1,27 @@
+use std::cmp::Ordering;
+use ImageConfig;
+use ImageScore;
+
+pub fn find_optimal_config(config: &Vec<ImageConfig>, threshold: f64) -> &ImageConfig {
+    let index = config.binary_search_by(|probe| {
+        // generate image with `probe` config
+        // run dssim against original
+        // return Ordering according to threshold
+        unimplemented!();
+    }).unwrap();
+
+    &config[index]
+}
+
+// rank based on min size with score above threshold
+pub fn rank_image_scores(scores: &Vec<ImageScore>, score_threshold: f64) -> Vec<ImageScore> {
```

Comment:
we can run in parallel the find_optimal_config for different formats and the results